src/index.js

```diff
@@ -157,13 +157,33 @@ async function getRDSValue(key, awsParameters) {
   return instances[0]
 }
 
+/**
+ * @param key the concatenated {stackName_logicalResourceId} of the CloudFormation to resolve physicalResourceId
+ * @param awsParameters parameters to pass to the AWS.CF constructor
+ * @returns {Promise.<String>} a promise for the resolved variable
+ * @see https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CloudFormation.html#describeStackResource-property
+ */
+async function getCFPhysicalResourceId(key, awsParameters) {
+  winston.debug(`Resolving a CloudFormation stack's PhysicalResourceId by the concatenated {stackName_logicalResourceId} ${key}`)
+  const cf = new AWS.CloudFormation({ ...awsParameters, apiVersion: '2014-10-31' })
+  const stackName = key.split('_')[0]
```

Comment:
I wonder if there is a more elegant way of passing all the necessary info in? At the least, we should probably do a bit error handling so that if someone makes a mistake the error message is clear

Reply:
@sup

> Yes you are right.
> At first I was wondering about changing the regex to accept "key" and "subkey" value, but finally I decided not to make a huge modification of main parsing function (also because we have already "service" and "subservice").
> Since "_" is the only special character in "\w+", and "_" is not in my convention of aws variable naming, I tried to use it for variable separation. 😄
> 
> But Indeed, I agree that there is a more elegant way, would it be more clear to add a symbol like "#" in regex? Thanks.
just my little view:
as "#" will keep the anchor meaning and avoid those who using '_' in their naming convention,
adding just another ':' level keep the same way of selecting something with this framework and be more easy to use, no ?